### PR TITLE
Major NextJs link cleanup

### PR DESCRIPTION
**Pull request by siggakogk**
# Remove `as`from next links

As of Next.js 9.5.3 the `as` prop is optional.
Cleaning up and simplifying next links 

> `as` - Optional decorator for the path that will be shown in the browser URL bar. Before Next.js 9.5.3 this was used for dynamic routes, check our previous docs to see how it worked

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/web/screens/ApiCatalogue/ApiCatalogue.tsx [hunk 1]
```diff
@@ -244,7 +244,7 @@ const ApiCatalogue: Screen<ApiCatalogueProps> = ({
               items={navigationItems}
               title={nn('linkThrounText')}
               titleLink={{
-                href: linkResolver('developerspage').as,
+                href: linkResolver('developerspage').href,
               }}
```
Comment (stefanhar) [inline apps/web/screens/ApiCatalogue/ApiCatalogue.tsx:248]: The wrapping object is technically not needed here, we might also write this like: ```titleLink={ linkResolver('developerspage')}```

File: apps/web/components/AboutLatestNews/AboutLatestNews.tsx [hunk 2]
```diff
@@ -123,7 +120,7 @@ const BigNewsItem = ({
         {news.title}
       </Text>
       <Text variant="intro">{news.intro}</Text>
-      <Link as={as} href={href}>
+      <Link href={href} passHref>
```
Comment (stefanhar) [inline apps/web/components/AboutLatestNews/AboutLatestNews.tsx:123]: We should probably apply `passHref` in island-UI to all links

File: apps/web/components/AboutLatestNews/AboutLatestNews.tsx [hunk 3]
```diff
@@ -61,7 +61,7 @@ export const AboutLatestNews: FC<LatestNewsProps> = ({
         {first && (
           <BigNewsItem
             news={first}
-            {...linkResolver('news', [first.slug])}
+            href={linkResolver('news', [first.slug]).href}
```
Comment (stefanhar) [inline apps/web/components/AboutLatestNews/AboutLatestNews.tsx:64]: We might also write all instances where the prop is `href` like this
```suggestion
            {...linkResolver('news', [first.slug])}
```

**Review by stefanhar — CHANGES_REQUESTED** [review]

File: apps/web/screens/ApiCatalogue/ApiCatalogue.tsx (hunk 1 above)
Comment (siggakogk) [reply apps/web/screens/ApiCatalogue/ApiCatalogue.tsx:248]: this object is unrelated to next.js so i don't want to be spreading the linkresolver object here, just cherrypicking the href string

**Review by stjanilofts — APPROVED** [review]

**Review by stefanhar — APPROVED** [review]
Awesome

**Pull request merged**